src/image/loading_displaying.js

```diff
@@ -598,106 +598,104 @@ function _createGif(
-      });
-
-      // Some GIFs are encoded so that they expect the previous frame
-      // to be under the current frame. This can occur at a sub-frame level
-      //
-      // Values :    0 -   No disposal specified. The decoder is
-      //                   not required to take any action.
-      //             1 -   Do not dispose. The graphic is to be left
-      //                   in place.
-      //             2 -   Restore to background color. The area used by the
-      //                   graphic must be restored to the background color.
-      //             3 -   Restore to previous. The decoder is required to
-      //                   restore the area overwritten by the graphic with
-      //                   what was there prior to rendering the graphic.
-      //          4-7 -    To be defined.
-      if (frameInfo.disposal === 2) {
-        // Restore background color
-        pImg.drawingContext.clearRect(
-          frameInfo.x,
-          frameInfo.y,
-          frameInfo.width,
-          frameInfo.height
-        );
-      } else if (frameInfo.disposal === 3) {
-        // Restore previous
-        pImg.drawingContext.putImageData(
-          prevFrameData,
-          0,
-          0,
-          frameInfo.x,
-          frameInfo.y,
-          frameInfo.width,
-          frameInfo.height
-        );
-      }
-    }
-
-    //Uses Netscape block encoding
-    //to repeat forever, this will be 0
-    //to repeat just once, this will be null
-    //to repeat N times (1<N), should contain integer for loop number
-    //this is changed to more usable values for us
-    //to repeat forever, loopCount = null
-    //everything else is just the number of loops
-    let loopLimit = gifReader.loopCount();
-    if (loopLimit === null) {
-      loopLimit = 1;
-    } else if (loopLimit === 0) {
-      loopLimit = null;
     }
+  }
 
-    // we used the pImg for painting and saving during load
-    // so we have to reset it to the first frame
-    pImg.drawingContext.putImageData(frames[0].image, 0, 0);
-
-    pImg.gifProperties = {
-      displayIndex: 0,
-      loopLimit,
-      loopCount: 0,
-      frames,
-      numFrames,
-      playing: true,
-      timeDisplayed: 0,
-      lastChangeTime: 0
-    };
+  //Uses Netscape block encoding
+  //to repeat forever, this will be 0
+  //to repeat just once, this will be null
+  //to repeat N times (1<N), should contain integer for loop number
+  //this is changed to more usable values for us
+  //to repeat forever, loopCount = null
+  //everything else is just the number of loops
+  let loopLimit = gifReader.loopCount();
+  if (loopLimit === null) {
+    loopLimit = 1;
+  } else if (loopLimit === 0) {
+    loopLimit = null;
   }
 
+  // we used the pImg for painting and saving during load
+  // so we have to reset it to the first frame
+  pImg.drawingContext.putImageData(frames[0].image, 0, 0);
+
+  pImg.gifProperties = {
```

Comment:
How do you feel about adding the if statement back just on this block here? If there's just one frame in the gif, if we don't set gifProperties, we can maybe save a few cpu cycles by avoiding trying to calculate what frame it should be on by treating it like a normal image from here on out.  At least I think so anyway, your new tests will help confirm it 🙂

Reply:
I think that should fix the test that's currently taking, which has the title "single frame gif should not have gifProperties". But if there are other complications, it's not the end of the world if they do have that, so we can talk about updating the test if we can't get it to work otherwise.

Reply:
Wow this is great feedback thanks! It's exactly why I was hoping to contribute :)

I totally agree and have submitted the conditional! But I'm kinda new to PRs so I'm not sure if they were added to this one...it looks like they were?

I took some time to look at where else gifProperties is being used but I think this is safe because the first frame is stored into the pImg, which is what eventually gets drawn. One thing to note is that this may prevent the ability add extra frames later (for example, to make use of the automatic GIF features), but that might not be intended anyways

Reply:
Right, I think we don't have any officially documented APIs for adding frames to imported gifs so I think we should be good!